blog/Getting Started with ANP.md

```diff
@@ -0,0 +1,148 @@
+
+1. There are certain traffic patterns in the cluster that, as a cluster admin, I want to always allow. For example, all namespaces should allow ingress traffic from a monitoring application in the cluster for compliance reasons, and all namespaces should allow egress toward kube-dns running in the kube-system namespace.
+
+2. As a cluster admin I want to change the default security model for my cluster, so that all inter-tenant traffic (traffic between namespaces with different tenant labels) is blocked by default. Namespace owners will need to use NetworkPolicies to explicitly allow known traffic.
+
+Both of the intents above however, are extremely difficult, if not impossible, to be captured by current K8s network policies. For example, many developers find themselves accidentally block required traffic (most commonly dns) when they apply network policies in their namespaces. This is due to the implicit deny nature of the network policy semantics: pods selected by network policies are automatically isolated except for ingress/egress traffic explicitly allowed. So, without any admin-facing policy APIs, to "ensure" (not really, we'll see why in a bit) that dns always works and monitoring probes are always permitted, cluster admins would have to apply the following specs in each and every namespace of the cluster:
+
+``` yaml
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy      # this doesn't work well
+metadata:
+  name: namespace-basic-allow
+  namespace: {{ every namespace in the cluster }}
+spec:
+  podSelector: {}
+  ingress:
+    - from:
+      - namespaceSelector:
+          matchlabels:
+            kubernetes.io/metadata.name: monitoring-ns
+  egress:
+    - to:
+      - namespaceSelector:
+          matchlabels:
+            kubernetes.io/metadata.name: kube-system
+        podSelector:
+          matchlabels:
+            app: kube-dns
+```
+
+There are a couple of problems with the above approach: 1. Namespace owners can easily override the NetworkPolicy rules by modifying them directly or simply deleting the policy, since the objects are namespace-scoped and owned by the namespace; 2. Namespace can be dynamically created and terminated. Cluster admins will need to utilize some external operator/controller to ensure that each namespace currently in the cluster is replicated with such NetworkPolicy; 3. There's simply no way for cluster admins to express intents like, I want to take a default security stance on certain types of traffic, but I want to also delegate the ultimate decision to each namespace owner.
+
+So why does the NetworkPolicy API fail to serve these admin use cases? Short answer: it was not designed for them to begin with.
+
+NetworkPolicy was designed around the idea of allowing the user deploying an application to control who can and cannot reach that application. It was not intended to be used for any other use cases, and its design reflects this:
```

Comment:
These were shamelessly copied from @danwinship's comment on the original KEP btw